lib/src/repo.rs

```diff
@@ -816,7 +820,7 @@ impl MutableRepo {
     /// docstring for `record_rewritten_commit` for details.
     pub fn set_rewritten_commit(&mut self, old_id: CommitId, new_id: CommitId) {
         assert_ne!(old_id, *self.store().root_commit_id());
-        self.rewritten_commits.insert(old_id, vec![new_id]);
+        self.parent_mapping.insert(old_id, vec![new_id]);
```

Comment:
nit: maybe better to delete or assert `self.divergent` entry?

Reply:
Good point. I made this function delete it.